app/src/main/java/org/example/ChessPieces/Bishop.java

```diff
@@ -0,0 +1,44 @@
+package org.example.ChessPieces;
+
+import org.example.BoardPosition;
+import org.example.BoardPosition.InvalidBoardPosition;
+
+public class Bishop extends ChessPiece {
+    Bishop(char piece) {
+        super(piece);
+    }
+
+    @Override
+    public boolean[][] getAllPossiblePositions(BoardPosition pos) {
```

Comment:
It would be cleaner to return a collection of positions, as opposed to a matrix of true/false.

So that the caller can do:
```
List<BoardPosition> possiblePositions = piece.getAllPossiblePositions(pos);
if (possiblePositions.contains(newPos)) {
  // move the piece
}
```

This is better because `possiblePositions.contains(newPos)` makes index mistakes impossible, for example: `possiblePositions[newPos.i][newPos.i]`

Reply:
I thought about this but also think lookup is faster from a 2D matrix. But given there's only 64 squares on the board, of which only a portion can be possible moves for a piece (queen will have the most), I agree that a cleaner/more workable code is a higher priority than lookup speed.

Reply:
Implemented.